File: src/2924.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    int findChampion(int n, vector<vector<int>>& edges) {
        int retVal = -1;

        vector<int> indegree(n, 0);
        for (auto edge : edges) {
            indegree[edge[1]] += 1;
        }

        for (int i = 0; i < n; ++i) {
            if (indegree[i] != 0) {
                continue;
            }

            if (retVal == -1) {
                retVal = i;
            } else {
                retVal = -1;
                break;
            }
        }

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        int n;
        vector<vector<int>> edges;
    };
    vector<subject> testData{{3, {{0, 1}, {1, 2}}}, {4, {{0, 2}, {1, 3}, {1, 2}}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: n = 3, edges = [[0,1],[1,2]]
     *  Output: 0
     *
     *  Input: n = 4, edges = [[0,2],[1,3],[1,2]]
     *  Output: -1
     */

    Solution cSolution;
    int answer;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: n = " << testData[i].n << ", edges = [";
        for (long unsigned int j = 0; j < testData[i].edges.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << "[";
            for (long unsigned int k = 0; k < testData[i].edges[j].size(); ++k) {
                cout << ((k == 0) ? "" : ",") << testData[i].edges[j][k];
            }
            cout << "]";
        }
        cout << "]\n";

        answer = cSolution.findChampion(testData[i].n, testData[i].edges);
        cout << "Output: " << answer << "\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}

File: src/2924.py
import logging
import sys
from bisect import bisect_left, bisect_right
from collections import Counter, defaultdict, deque
from functools import cache, lru_cache
from heapq import heapify, heappop, heappush
from itertools import accumulate
from typing import List, Optional


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        retVal = -1

        indegree = [0] * n
        for edge in edges:
            indegree[edge[1]] += 1

        for i in range(n):
            if indegree[i] != 0:
                continue

            if retVal == -1:
                retVal = i
            else:
                retVal = -1
                break

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for n, edges in zip([3, 4], [[[0, 1], [1, 2]], [[0, 2], [1, 3], [1, 2]]]):
            # /* Example
            #  *  Input: n = 3, edges = [[0,1],[1,2]]
            #  *  Output: 0
            #  *
            #  *  Input: n = 4, edges = [[0,2],[1,3],[1,2]]
            #  *  Output: -1
            #  */
            logging.info("Input: n = %s, edges = %s", n, edges)

            retVal = pSolution.findChampion(n, edges)
            logging.info("Output: %s", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)
